### PR TITLE
Implement $push, fix leaking, add more guards for updates, small cleanup

### PR DESCRIPTION
**Pull request by scottrogowski**

### CODE REVIEW

File: tests/test_mongita.py [hunk 1]
```diff
@@ -635,6 +635,71 @@ def test_update_many(client_class):
...
 
 
+@pytest.mark.parametrize("client_class", CLIENTS)
+def test_push(client_class):
+    client, coll, imr = setup_many(client_class)
+    assert len(coll.find_one({"name": "Meercat"})["continents"]) == 1
+    coll.update_one({"name": "Meercat"}, {"$push": {"continents": "foobar"}})
+    new_meercat_continents = coll.find_one({"name": "Meercat"})["continents"]
+    assert len(new_meercat_continents) == 2
+    assert set(new_meercat_continents) == {"foobar", "AF"}
```
Comment (Kilo59) [inline tests/test_mongita.py:645]: Only thing I would say about this is that by using testing a `set` we could be potentially be making changes to the array order and not realize it. Which I believe standard Mongo preserves array order.

**Review by Kilo59 — APPROVED** [review]
1 minor comment about array ordering but it all looks good to me.

I may submit a very small PR to add some $push examples to the Readme.

File: tests/test_mongita.py (hunk 1 above)
Comment (scottrogowski) [reply tests/test_mongita.py:645]: You make a good point. Most Mongita unit tests make use of sets but regular MongoDB does preserve order and using sets might hide ordering bugs.

**Comment by scottrogowski** [comment]
@Kilo59 merged. Will try to release this with the other PRs that are in flight right now. I think they are all pretty close so should be soon

**Pull request merged**